rust/src/transformations/clamp/make_clamp.tex

```diff
@@ -1,65 +1,75 @@
+    Assume the input is a member of \texttt{input\_domain}'s row domain.  
+    Therefore, by \ref{line:assert-non-null}, the input is non-null. 
+    In addition, since \texttt{Bounds.new\_closed} did not raise an exception, then by the definition of \texttt{Bounds.new\_closed}, the bounds are non-null. 
+    
+    \label{lemma:row-by-row-precondition} 
+    The preconditions of \texttt{make\_clamp} and pseudocode definition (line \ref{line:def}) ensure that the type preconditions of \texttt{make\_row\_by\_row\_fallible} are satisfied.  
+    The remaining preconditions of \texttt{make\_row\_by\_row\_fallible} are: 
+    \begin{itemize} 
+        \item Errors from \texttt{row\_function} are data-independent.
         \item \texttt{row\_function} has no side-effects.
-        \item If the input to \texttt{row\_function} is a member of \texttt{input\_domain}'s row domain, then the output is a member of \texttt{output\_row\_domain}.
+        \item If the input to \texttt{row\_function} is a member of \texttt{input\_domain}'s row domain, then the output is a member of \texttt{output\_row\_domain}. 
     \end{itemize}
-
-    The first precondition is satisfied by the definition of \texttt{clamper} (line \ref{line:clamper}) in the pseudocode.
-
-    For the second precondition, assume the input is a member of \texttt{input\_domain}'s row domain. 
-    Therefore, by \ref{line:assert-non-null}, the input is non-null.
-    In addition, since \texttt{Bounds.new\_closed} did not raise an exception, then by the definition of \texttt{Bounds.new\_closed}, the bounds are non-null.
-    Thus, by the definition of \rustdoc{traits/trait}{ProductOrd}, the preconditions of \texttt{total\_clamp} are satisfied, so the output is within the bounds.
-    Therefore, the output is a member of \texttt{output\_row\_domain}.
-\end{proof}
-
-We now prove the postcondition of \texttt{make\_clamp}.
-\begin{proof}
-By \ref{lemma:row-by-row-precondition}, the preconditions of \rustdoc{transformations/manipulation/fn}{make\_row\_by\_row\_fallible} are satisfied.
-Thus, by the definition of \\\texttt{make\_row\_by\_row\_fallible}, the output is a valid transformation.
-\end{proof}
-
+    
+    \item By the definition of \rustdoc{traits/trait}{ProductOrd}\texttt{.total\_clamp}, 
+    \texttt{total\_clamp} won't raise because \texttt{arg} and both bounds are not null.
+    Since \texttt{total\_clamp} is the only potential source of errors, and it cannot throw an error,
+    there are no runtime errors in \texttt{row\_function}, satisfying the first remaining precondition.
+
+    The second remaining precondition is satisfied by the definition of \texttt{clamper} (line \ref{line:clamper}) in the pseudocode.
+    The function uses \rustdoc{traits/trait}{ProductOrd}\texttt{.total\_clamp}, and by its definition there are no side-effects.
+ 
+    For the last remaining precondition, 
+    by the postcondition/proof definition of \rustdoc{traits/trait}{ProductOrd}\texttt{.total\_clamp},
```

Comment:
I don't know why I didn't see it before, but it's there: `where min <= max`. The restriction on min/max is a precondition, whereas we can claim that the output is not an error because it only throws an error if any of the args are null, and none are.